output/transform/merge_partition_sizes.go

```diff
@@ -0,0 +1,56 @@
+package transform
+
+import (
+	snapshot "github.com/pganalyze/collector/output/pganalyze_collector"
+	"github.com/pganalyze/collector/state"
+)
+
+func mergePartitionSizes(s snapshot.FullSnapshot, newState state.PersistedState, ts state.TransientState, databaseOidToIdx OidToIdx) snapshot.FullSnapshot {
+	for idx, info := range s.IndexInformations {
+		rel := s.RelationInformations[info.RelationIdx]
+		if !rel.HasParentRelation || rel.PartitionBoundary == "" {
+			continue
+		}
+		for parentIdx, pi := range s.IndexInformations {
+			if pi.RelationIdx != rel.ParentRelationIdx {
+				continue
+			}
+			if info.IndexType == pi.IndexType && info.IsUnique == pi.IsUnique && intArrayEqual(info.Columns, pi.Columns) {
```

Comment:
This will spuriously match different expression indexes on the same columns, but I think that's okay: I don't see a great way to avoid that without a considerably more expensive check, and it's unlikely we'll run into that in practice.